pkg/controllers/management/catalog/controller.go

```diff
@@ -79,6 +80,9 @@ func Run(ctx context.Context, cacheRoot string, refreshInterval int, management
 	clusterCatalogController := management.Management.ClusterCatalogs("").Controller()
 	clusterCatalogController.AddHandler(ctx, "clusterCatalog", m.ClusterCatalogSync)
 
+	go m.DeleteOldTemplateContent()
+	go m.DeleteBadCatalogTemplates()
```

Comment:
need to address the scenario where error happens and the code will never be executed until server restarts